### PR TITLE
Fix config assets/storage_duration

### PR DESCRIPTION
**Pull request by andrii-suse**
Add CURRENT to list of known keywords in Setup.pm, otherwise it is ignored here:

https://github.com/os-autoinst/openQA/blob/d93df0ea5bcab9b4b8e6967a9b5e21a0c04c7e59/lib/OpenQA/Setup.pm#L245

### CODE REVIEW

File: lib/OpenQA/Setup.pm [hunk 1]
```diff
@@ -202,7 +202,7 @@ sub read_config {
             untracked_assets_storage_duration => 14,
         },
         'assets/storage_duration' => {
-            # intentionally left blank for overview
+            CURRENT => 30,
```
Comment (okurz) [inline lib/OpenQA/Setup.pm]: I don't think "CURRENT" is something that should be added by default in openQA. I consider the "CURRENT" assets a SUSE-specific workaround with no place in upstream openQA.

File: t/config.t [hunk 2]
```diff
@@ -157,6 +162,28 @@ subtest 'Test configuration override from file' => sub {
         ],
         'referers parsed correctly'
     );
+    my $untracked_assets_patterns = $app->config->{'assets/storage_duration'};
+
+    ok(ref $untracked_assets_patterns eq ref {}, "failed to read config for assets/storage_duration");
```
Comment (okurz) [inline t/config.t]: the test string "failed to read" reads a bit weirdly. Can you rephrase this so that it would not confuse a reader of the test output?

File: t/config.t [hunk 3]
```diff
@@ -157,6 +162,28 @@ subtest 'Test configuration override from file' => sub {
...
+    ok(scalar %$untracked_assets_patterns, "config is empty for assets/storage_duration");
```
Comment (okurz) [inline t/config.t]: `ok(scalar %hash` is IMHO the weirdest possible way to check for an empty hash :D

File: t/config.t [hunk 4]
```diff
@@ -157,6 +162,28 @@ subtest 'Test configuration override from file' => sub {
...
+    for my $pattern (keys %$untracked_assets_patterns) {
+        if ($pattern eq 'CURRENT') {
+            $current_is_here = 1;
+            $current_value   = $untracked_assets_patterns->{$pattern};
+            if ('SLE-12-SP5-HA-POOL-x86_64-CURRENT-Media1' =~ $pattern) {
+                $asset_name_matches = 1;
+            }
+        }
+    }
+    ok($current_is_here,     'CURRENT is configured');
```
Comment (okurz) [inline t/config.t]: I think it would look nicer to just use `grep` instead of a for-loop for this check

File: lib/OpenQA/Setup.pm (hunk 1 above)
Comment (andrii-suse) [reply lib/OpenQA/Setup.pm]: So what Is your suggestion about fixing current urgent problem? Fork OpenQA? maybe we provide proper solution later?

File: t/config.t (hunk 2 above)
Comment (kraih) [reply t/config.t]: What is this test supposed to do? Why does a hashref signal a failed read? Why `eq ref {}` instead of `eq 'HASH'`?

File: t/config.t (hunk 3 above)
Comment (kraih) [reply t/config.t]: Yea, just don't do that. Check the number of keys instead.

File: t/config.t [hunk 5]
```diff
@@ -157,6 +162,28 @@ subtest 'Test configuration override from file' => sub {
         ],
         'referers parsed correctly'
     );
+    my $untracked_assets_patterns = $app->config->{'assets/storage_duration'};
+
+    ok(ref $untracked_assets_patterns eq ref {}, "failed to read config for assets/storage_duration");
+    ok(scalar %$untracked_assets_patterns, "config is empty for assets/storage_duration");
+
+    return unless (ref $untracked_assets_patterns eq ref {} and scalar %$untracked_assets_patterns);
```
Comment (kraih) [inline t/config.t]: What do you gain from exiting early in case of test failure? Is there no chance later tests can reveal more information?

File: t/config.t (hunk 4 above)
Comment (kraih) [reply t/config.t]: Agreed, this is way more verbose than it has to be.

File: lib/OpenQA/Setup.pm (hunk 1 above)
Comment (Martchus) [reply lib/OpenQA/Setup.pm]: Simply adjust the loop/assignments at the end of this function to handle that case (allow arbitrary keys within 'assets/storage_duration' section).

**Comment by andrii-suse** [comment]
It was decided to implement a way to read hash from ini instead

**Comment by andrii-suse** [comment]
should be fixed by #2715
